Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp

```diff
@@ -218,8 +219,19 @@ void TextureMapperLayer::paintSelf(TextureMapperPaintOptions& options)
         solidColorLayer.setColor(m_state.solidColor);
         contentsLayer = &solidColorLayer;
     }
-    if (!contentsLayer)
+
+    if (!contentsLayer) {
+        // Use the damage information we received from the CoordinatedGraphicsLayer
+        // Here we ignore the targetRect parameter as it should already have
+        // been covered by the damage tracking in setNeedsDisplay/setNeedsDisplayInRect
+        // calls from CoordinatedGraphicsLayer.
+        for (auto& region : m_damaged)
+            recordDamage(region, transform, options);
```

Comment:
`recordDamage` conditionally calls `m_damaged = {}` which may lead to elements being removed while iterating

Reply:
done